src/components/user/ProposalsPeopleTable.tsx

```diff
@@ -0,0 +1,515 @@
+                  aria-label="close"
+                  color="inherit"
+                  size="small"
+                  onClick={() => {
+                    setDisplayError(false);
+                  }}
+                >
+                  <CloseIcon fontSize="inherit" />
+                </IconButton>
+              }
+            >
+              We cannot find that email. Please check the spelling and if the
+              user has registered with us or has the correct privacy settings to
+              be found by this search.
+            </Alert>
+          )}
+          <Form className={classes.email}>
+            <Field
+              name="email"
+              label="E-mail"
+              type="email"
+              component={TextField}
+              margin="normal"
+              fullWidth
+              flex="1"
+              data-cy="email"
+            />
+            <Button
+              data-cy="findUser"
+              variant="contained"
+              color="primary"
+              type="submit"
+              className={classes.inviteButton}
+            >
+              Find User
+            </Button>
+          </Form>
+        </div>
+      </>
+    );
+  };
+
+  return (
+    <Formik
+      initialValues={{
+        email: '',
+      }}
+      initialErrors={{
+        email: 'test',
+      }}
+      onSubmit={async (values, { setFieldError, setFieldValue }) => {
+        // If there is an email and it has not already been searched
+        if (values.email && !tableEmails.includes(values.email)) {
+          setLoading(true);
+          const userDetails = await getUserByEmail(values.email, sendRequest);
+
+          if (!userDetails) {
+            setDisplayError(true);
+            setFieldError('email', 'Please enter valid email address');
+            setLoading(false);
+
+            return;
+          }
+
+          if (props.selectedUsers?.includes(userDetails.id)) {
+            setFieldError('email', 'User is already on the proposal');
+            setLoading(false);
+
+            return;
+          }
+
+          if (!invitedUsers.some((user) => user.id === userDetails.id)) {
+            //Add users to the table
+            setInvitedUsers([userDetails].concat(invitedUsers));
+            setTableEmails(tableEmails.concat([values.email]));
+            setFieldValue('email', '');
+
+            //If we are selecting multiple users add the user as pre selected.
+            if (props.selection)
+              setSelectedParticipants(
+                selectedParticipants.concat([userDetails])
+              );
+
+            //This is the recommend way to refresh a table but it is unsupported in typescript and may not be supported anytime soon
+            //See: Refresh Data Example https://material-table.com/#/docs/features/remote-data
+            //and see: https://github.com/mbrn/material-table/issues/1752
+            // @ts-ignore
+            tableRef.current.onQueryChange();
```

Comment:
I could not find a better solution for this. If anyone else can I would happy to change it.

Reply:
@TCMeldrum I think you can use something like this https://github.com/UserOfficeProject/user-office-frontend/blob/develop/src/components/user/PeopleTable.tsx#L244 if I understand the need correctly it is just to refresh the table data after you submit the form.